### PR TITLE
Allow customizing gacela files naming

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Allow to rename the gacela files (Factory, Config, DependencyProvider).

## 🔖 Changes

- Create a feature test about `UsingCustomResolvableTypes`
- Allow overriding the gacela resolvable types. For example:
```php
public function overrideResolvableTypes(): array
{
    return [
        'Factory' => 'FactCustom',
        'Config' => 'ConfCustom',
        'DependencyProvider' => 'DepProvCustom',
    ];
}
```

### CODE REVIEW

**Comment by Tito-Kati** [comment]
Missing some tests for custom names on configuration and providers.

**Comment by Chemaclass** [comment]
@Tito-Kati the feature test includes/tests all gacela files. It starts from the Facade and loads it's Factory, Config and DependencyProvider to generate the expected assertion with the different values coming from the custom classes.

File: src/Framework/ClassResolver/AbstractClassResolver.php [hunk 1]
```diff
@@ -180,6 +180,16 @@ private function getClassNameFinder(): ClassNameFinderInterface
         return $this->classNameFinder;
     }
 
+    /**
+     * Allow overriding gacela resolvable types.
+     */
+    private function getFinalResolvableType(): string
+    {
+        $gacelaFile = $this->getGacelaConfigFile()->getOverrideResolvableTypes();
```
Comment (JesusValera) [inline src/Framework/ClassResolver/AbstractClassResolver.php:188]:
```suggestion
        $overrideResolvableTypes = $this->getGacelaConfigFile()->getOverrideResolvableTypes();
```

File: src/Framework/ClassResolver/AbstractClassResolver.php [hunk 2]
```diff
@@ -180,6 +180,16 @@ private function getClassNameFinder(): ClassNameFinderInterface
...
+
+        return $gacelaFile[$this->getResolvableType()] ?? $this->getResolvableType();
```
Comment (JesusValera) [inline src/Framework/ClassResolver/AbstractClassResolver.php:190]:
```suggestion
        return $overrideResolvableTypes[$this->getResolvableType()] ?? $this->getResolvableType();
```

**Review by JesusValera — APPROVED** [review]

**Pull request merged**